app/models/shipit/task.rb

```diff
@@ -57,7 +58,7 @@ def current
         task.started_at ||= Time.now.utc
       end
 
-      before_transition any => %i(success failed error timedout validating) do |task|
+      before_transition any => END_STATUSES.map(&:to_sym) do |task|
```

Comment:
Why not https://github.com/Shopify/shipit-engine/pull/849#discussion_r250537738 ?

It's much simpler, and much more forward compatible.